pkg/ottl/ottlfuncs/func_to_time.go

```diff
@@ -0,0 +1,62 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type ToTimeArguments[K any] struct {
+	Time   ottl.StringGetter[K] `ottlarg:"0"`
+	Format ottl.StringGetter[K] `ottlarg:"1"`
+}
+
+func NewToTimeFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("toTime", &ToTimeArguments[K]{}, createToTimeFunction[K])
```

Comment:
OTTL requires that converter functions start with a capital letter.  To be consistent with our other type-creating convertes such as `Int`, `SpanID`, and `TraceID`, let's call this function `Time` and update the file's function names accordingly.